### PR TITLE
Hyperdrive API for Spark 3

### LINKED ISSUE #1712: Hyperdrive API for Scala 2.12 for Enceladus@Spark3
## Background
~Hyperdrive API is not yet released for Scala 2.12.~ Let's integrate it when it comes out in the Spark3 version of Enceladus.

Hyperdrive API 4.2.1 for Scala 2.12 now exists.

### PR DESCRIPTION
**Pull request by dk1844**
As Hyperdrive for Spark 2.12 has been released, this PR reenables it for Enceladus@Spark3.

~`HyperConformanceMappingIntegrationSuite` depends on #1744 (when naively merged with the would-be #1744, the test works ok)~
Otherwise, the unit and integration tests pass.

Closes #1712 .

### CODE REVIEW

**Review by AdrianOlosutean — COMMENTED** [review]
I may have noticed, but are there any changes from how it was before or did it work out of the box?

**Review by AdrianOlosutean — DISMISSED** [review]

**Comment by dk1844** [comment]
> I may have noticed, but are there any changes from how it was before or did it work out of the box?

No, no changes needed except for the dependency update.

**Review by AdrianOlosutean — DISMISSED** [review]

**Review by AdrianOlosutean — APPROVED** [review]

**Comment by AdrianOlosutean** [comment]
`HyperConformanceMappingIntegrationSuite` should be enabled

**Review by AdrianOlosutean — APPROVED** [review]

**Comment by dk1844** [comment]
The unit and extended integration testing was concluded to be sufficient - "no test needed" label applied.

File: spark-jobs/src/test/scala/za/co/absa/enceladus/conformance/streaming/HyperConformanceMappingIntegrationSuite.scala [hunk 1]
```diff
@@ -13,28 +13,26 @@
...
-//    val conformed = spark.read
-//      .textFile("src/test/testData/multipleMapping/conformed_multiple_mapping.json")
-//      .collect().mkString("\n")
-//    val returned = df.toJSON.collect().mkString("\n")
-//    assertResult(returned)(conformed)
-//  }
-//}
+class HyperConformanceMappingIntegrationSuite extends AnyFunSuite with StreamingFixture with MultipleMappingFixture {
+
+  // todo dependent on resolving #1744 udf for spark 3
```
Comment (benedeki) [inline spark-jobs/src/test/scala/za/co/absa/enceladus/conformance/streaming/HyperConformanceMappingIntegrationSuite.scala:24]: What about this TODO?

File: spark-jobs/src/test/scala/za/co/absa/enceladus/conformance/interpreter/InterpreterSuite.scala [hunk 2]
```diff
@@ -86,7 +86,7 @@ class InterpreterSuite extends AnyFunSuite with SparkTestBase with BeforeAndAfte
     spark.disableControlMeasuresTracking()
 
 
-    eventually(timeout(scaled(10.seconds)), interval(scaled(500.millis))) {
+    eventually(timeout(scaled(15.seconds)), interval(scaled(500.millis))) {
```
Comment (benedeki) [inline spark-jobs/src/test/scala/za/co/absa/enceladus/conformance/interpreter/InterpreterSuite.scala:89]: Just curious: why the change?

Comment (dk1844) [reply spark-jobs/src/test/scala/za/co/absa/enceladus/conformance/interpreter/InterpreterSuite.scala:89]: It can be quite slow (on battery, e.g.). Mainly on Jenkins, for which I increased the scale factor, but to make sure there is enough time.

File: spark-jobs/src/test/scala/za/co/absa/enceladus/conformance/streaming/HyperConformanceMappingIntegrationSuite.scala (hunk 1 above)
Comment (dk1844) [reply spark-jobs/src/test/scala/za/co/absa/enceladus/conformance/streaming/HyperConformanceMappingIntegrationSuite.scala:24]: This means that it can only be enabled (`ignore` -> `test`) after #1744 is merged in.

Comment (benedeki) [reply spark-jobs/src/test/scala/za/co/absa/enceladus/conformance/streaming/HyperConformanceMappingIntegrationSuite.scala:24]: Believe this can be now reverted.

**Review by benedeki — COMMENTED** [review]
The dependency has been merged, I think we can proceed with this one now.
But @AdrianOlosutean  had some changes reverted back in #1744 to be done here.

File: spark-jobs/src/test/scala/za/co/absa/enceladus/conformance/streaming/HyperConformanceMappingIntegrationSuite.scala (hunk 1 above)
Comment (dk1844) [reply spark-jobs/src/test/scala/za/co/absa/enceladus/conformance/streaming/HyperConformanceMappingIntegrationSuite.scala:24]: Done.

**Review by AdrianOlosutean — APPROVED** [review]
